src/pages/routes/components/RouteComponent.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import { toast } from 'react-toastify';
+
+import Button from '../../../components/Button.js';
+import options from './RouteOptions.js';
+import { routesDistance } from './RoutesDistance.js';
+
+const RouteComponent = ({formTitle, formAction, data}) => {
+        const inputClassStyles = 'rounded-sm px-3 py-4 mb-5 bg-[#EFEFEF]';
+
+        const [destination1, setDestination1] = useState(null);
+        const [destination2, setDestination2] = useState(null);
+        const [distance, setDistance] = useState(null);
+
+        const handleSubmit = (e) => {
+            e.preventDefault();
+            formAction(destination1, destination2, distance)
+        }
+
+        const distanceValue = (destination1, destination2) => {
+            if (destination1 && destination2) {
+                if(destination1 == destination2) {
+                    toast("Destination one and Destination two can't match", {type: "error"})
+                    return
+                }
+               const selectedRoute =   routesDistance.filter((entry) => {
+                    return (entry.route.includes(destination1) && entry.route.includes(destination2));
+                });
+                if (selectedRoute) {setDistance(selectedRoute[0].distance) 
+                    return
+                }
+            }
+        };
+        
+        useEffect(() => {
+            distanceValue(destination1, destination2);
+        }, [destination2, destination2]);
+
```

Comment:
Make sure that the form validation is working as expected so far when you match destination 1 and destination 2 you get the error but when you choose destination 2 first validation doesn't work and a route without distance will be registered. and that's the same case with updating routes